### PR TITLE
Feature: add utils for keras saving/loading format detection

### PR DESCRIPTION
**Pull request by howl-anderson**
Add utils for Keras saving/loading format detection

### CODE REVIEW

**Comment by howl-anderson** [comment]
@k-w-w Please take a look. Thank you!

**Comment by k-w-w** [comment]
Sorry for the extremely late review! Thanks for adding this PR, can you rebase onto master? There shouldn't be too many merge conflicts.

Also, the TODO(b/130258301) refers to detecting whether the model is subclassed (rather than created with the functional/Sequential APIs), so it should be left in the code.

**Review by k-w-w — CHANGES_REQUESTED** [review]
(see above comment)

**Comment by howl-anderson** [comment]
@k-w-w Sure! I will update this PR later.

**Comment by howl-anderson** [comment]
@k-w-w I already update the PR, please take a look!

File: tensorflow/python/keras/saving/save.py [hunk 1]
```diff
@@ -44,6 +44,24 @@
 _KERAS_SAVED_MODEL_STILL_EXPERIMENTAL = True
 
 
+def _could_write_as_hdf5(filepath) -> bool:
+  """detect if `filepath` is a h5py.File object or its extension belongs to HDF5 extensions
```
Comment (k-w-w) [inline tensorflow/python/keras/saving/save.py:48]: nit: capitalize detect, and this should end with a period. 

Also, include args + returns

File: tensorflow/python/keras/saving/save.py [hunk 2]
```diff
@@ -44,6 +44,24 @@
...
+  """
+  file_is_h5_object = h5py is not None and isinstance(filepath, h5py.File)
+  implicit_h5_format = os.path.splitext(filepath)[1] in _HDF5_EXTENSIONS
+
+  return file_is_h5_object or implicit_h5_format
+
+
+def _could_read_as_hdf5(filepath) -> bool:
+  """detect if `filepath` is a h5py.File object or an hdf5 file
```
Comment (k-w-w) [inline tensorflow/python/keras/saving/save.py:57]: same here

**Review by k-w-w — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/saving/save.py (hunk 2 above)
Comment (howl-anderson) [reply tensorflow/python/keras/saving/save.py:57]: Thanks, I will update this PR later!

File: tensorflow/python/keras/saving/save.py (hunk 1 above)
Comment (howl-anderson) [reply tensorflow/python/keras/saving/save.py:48]: Thanks, I will update this PR later!

**Comment by gbaned** [comment]
@howl-anderson  Any update on this PR? Please. Thanks!

**Comment by howl-anderson** [comment]
@gbaned I have been unexpectedly busy in recent days and I will update this PR early next week.

**Comment by gbaned** [comment]
@howl-anderson Any update on this PR? Please. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by gbaned** [comment]
@howl-anderson Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!
